Library/Homebrew/cmd/tap.rb

```diff
@@ -27,11 +27,6 @@ def tap_args
         assumptions, so taps can be cloned from places other than GitHub and
         using protocols other than HTTPS, e.g. SSH, git, HTTP, FTP(S), rsync.
       EOS
-      switch "--full",
-             description: "Convert a shallow clone to a full clone without untapping. Taps are only cloned as "\
-                          "shallow clones if `--shallow` was originally passed."
-      switch "--shallow",
```

Comment:
Given the comments about deprecation path (which I agree with): let's leave these options in place but make/document them as a no-op. Add a commented-out `# odeprecated "`brew tap --full`" for each so we deprecate, disable and remove them on the usual minor release cycle. Shout if any of this was badly explained and you need help!